.instrumentation_generator/test/instrumentation_generator_test.rb

```diff
@@ -0,0 +1,84 @@
+# test/instrumentation_generator_test.rb
+
+require 'minitest/autorun'
+require 'mocha/minitest'
+require 'thor'
+require_relative '../instrumentation_generator'
+
+class InstrumentationGeneratorTest < Minitest::Test
+  def setup
+    @instrumentation_name = 'new_instrumentation'
+    @generator = InstrumentationGenerator.new([@instrumentation_name])
+
+    @generator.stubs(:template)
+    @generator.stubs(:insert_into_file)
+    @generator.stubs(:puts)
+  end
```

Comment:
Since running the generator creates files, I think the files should be deleted between the tests to make sure we have a clean slate for each test. 

Like the `setup` method you added, [minitest also has a `teardown` method.](https://www.rubydoc.info/stdlib/minitest/MiniTest%2FUnit%2FTestCase:teardown)

Maybe in `teardown` we could delete the directory we expect to get created if it exists?